src/Microsoft.Framework.Runtime/Servicing/ServicingBreadcrumbs.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using NuGet;
+
+namespace Microsoft.Framework.Runtime.Servicing
+{
+    internal class ServicingBreadcrumbs
+    {
+        private readonly string _breadcrumbsFolder;
+
+        public ServicingBreadcrumbs()
+            : this(ResolveBreadcrumbsFolder())
+        {
+        }
+
+        internal ServicingBreadcrumbs(string breadcrumbsFolder)
+        {
+            // If the directory doesn't exist, don't create it because it
+            // needs special permissions on it
+            _breadcrumbsFolder = Directory.Exists(breadcrumbsFolder) ? breadcrumbsFolder : null; ;
```

Comment:
I thought about that and I can definitely create it from code but I am worried about the permissions on it. If the code runs under a limited account, it cannot restrict the delete permissions to admins only. Also, in `%programdata%\Microsoft` only admins can write by default so the script that creates the folder must run elevated